core/io/SCsub

```diff
@@ -5,3 +5,4 @@ Import('env')
 env.add_source_files(env.core_sources, "*.cpp")
 
 Export('env')
+env.Append(CCFLAGS="-DZSTD_STATIC_LINKING_ONLY")
```

Comment:
I'm not sure how it works exactly in SCons, but I think it makes more sense to modify `env` before the `Export('env')` call (though in practice I'm not sure it changes anything, and I'm not even sure this exporting is needed, but it's for the sake of readability :)).

Reply:
Is it needed here though? You're setting this flag already in `thirdparty/zstd/SCsub`.

Reply:
Some function definitions in zstd.h are hidden behind that ifdef, and since zstd's per-module definition won't affect core/io code, yes, it is necessary

Reply:
Well we end up with `-DZSTD_STATIC_LINKING_ONLY -DZSTD_STATIC_LINKING_ONLY` in the CCFLAGS, it seems to be overkill. Both SCsub impact the same env.

And it's not just to be nitpicking, our build command line is very long already and makes logs of several megabytes:
```
g++ -o modules/gdscript/gd_editor.x11.tools.64.o -c -g3 -DDEBUG_ENABLED -DDEBUG_MEMORY_ENABLED -pipe -w -DZSTD_STATIC_LINKING_ONLY -DZSTD_STATIC_LINKING_ONLY -DFREETYPE_ENABLED -DRECAST_ENABLED -DSVG_ENABLED -DHAVE_MNTENT -DDEBUG_MEMORY_ALLOC -DSCI_NAMESPACE -DALSA_ENABLED -DJOYDEV_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DOPENGL_ENABLED -DGLES2_ENABLED -DGLES_OVER_GL -DPTRCALL_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -DGLAD_ENABLED -DGLES_OVER_GL -Icore -Icore/math -Ieditor -Idrivers -I. -I/usr/include/freetype2 -I/usr/include/libpng12 -I/usr/include/alsa -Iplatform/x11 -Ithirdparty/zstd -Ithirdparty/zstd/common -Ithirdparty/glad -Ithirdparty/recastnavigation/Recast -Ithirdparty/recastnavigation/Recast/Include -Ithirdparty/nanosvg modules/gdscript/gd_editor.cpp
```
So anything we can do without impacting CCFLAGS is best (e.g. using a custom zstd_godot_defs.h file which would define the flags we need if there were many - maybe not necessary here as long as it's only one).